docs: add zenodo export docs (#4480)
SwissDataScienceCenter/renku: docs/docs/10-users/40-data/guides/20-export-data-connector-to-zenodo.md

---
title: Publish data from a data connector to Zenodo
---

# Publish data from a data connector to Zenodo

You can export files from a Renku data connector directly to a draft deposit in [Zenodo](https://zenodo.org/). Renku creates the deposit in the background with basic metadata. You can then review, complete, and publish the draft in Zenodo.

:::info

Renku does not publish the deposit automatically. After Renku creates the draft deposit, you must finalize the metadata and publish it in Zenodo.

:::

## Before you begin

Make sure that:

- You have access to the data connector and to the files you want to publish.
- You have a Zenodo account.
- Your Renku account is connected to Zenodo. See [Step 1: activate the Zenodo integration](#step-1-activate-the-zenodo-integration).

## Step 1: activate the Zenodo integration

1. Open the Renku **Integrations** page.
2. Find the **Zenodo** integration.
3. Click **Connect**.
4. Follow the Zenodo authorization flow and grant Renku the requested permissions.

> 5. When you return to Renku, confirm that the Zenodo integration is shown as active.

Reviewer: Minor UI change: only active integrations appear on the page. For the others, users need to click on the "Plus" button

```suggestion
2. Click on the Add button and search for the **Zenodo** integration.
3. Click **Connect**.
4. Follow the Zenodo authorization flow and grant Renku the requested permissions.
5. When you return to Renku, confirm that the Zenodo integration is shown as active.
```

You only need to do this once per Renku account, unless the integration is disconnected or the authorization expires.

## Step 2: export files from the data connector

1. Open the project, group, or user page that contains the data connector.
2. Click the data connector to open its side panel.
3. Open the data connector drop-down menu.
4. Select the option to export or publish the data connector files to Zenodo.
5. Enter the path inside the data connector that contains the files you want to publish.

> 6. Confirm the export.

Reviewer: The menu is now available on the data connector listing too

```suggestion
2. Click the data connector drop-down menu.
3. Select the option to export or publish the data connector files to Zenodo.
4. Enter the path inside the data connector that contains the files you want to publish.
5. Confirm the export.
```

The next release will include the data connector's full pages.  Perhaps we can already add here:
```
If you are on a data connector page, the menu to export files is on the Settings tab
```

Reviewer: On second thought, we can still leave the point on clicking to open the side panel. Users will see better feedback on the ongoing export.

I would still add the part on "data connector page --> settings tab"

### Directory structure is not preserved

It is not possible to preseve a directory structure in Zenodo deposits. If the
selected path contains a directory structure that is important for understanding
or reusing the data, first create an archive file, such as a `.zip` or
`.tar.gz`, with the structure you want to preserve. Then export the path that
contains that archive file.

For example, instead of exporting this structure directly:

```text
results/
├── raw/
│   └── measurements.csv
├── processed/
│   └── summary.csv
└── README.md
```

create an archive such as `results.zip` and export the path containing `results.zip`.

## What happens after you start the export

After you confirm the export:

1. Renku uploads the selected files to Zenodo in the background.
2. Zenodo creates a draft deposit.
3. Renku adds basic metadata to the draft deposit.
4. You review and complete the draft deposit in Zenodo.
5. You publish the deposit from Zenodo when it is ready.

Depending on the number and size of files, the upload can take some time. You can continue working in Renku while the export runs.

## Complete and publish the deposit in Zenodo

> When the upload is complete, open the draft deposit in Zenodo and review it before publishing. In Zenodo, you should:

Reviewer: ```suggestion
When the upload is complete, the deposit export section will show a URL linking to the draft deposit in Zenodo for opening and reviewing the publishing. In Zenodo, you should:
```

Feel free to reword; I want to add the information about "you can stay in RenkuLab until we show you the link"
